x-pack/plugin/core/src/main/java/org/elasticsearch/snapshots/SourceOnlySnapshot.java

```diff
@@ -51,7 +51,7 @@
 import java.util.function.Supplier;
 
 import static org.apache.lucene.codecs.compressing.CompressingStoredFieldsWriter.FIELDS_EXTENSION;
-import static org.apache.lucene.codecs.compressing.CompressingStoredFieldsWriter.FIELDS_INDEX_EXTENSION;
+import static org.apache.lucene.codecs.compressing.CompressingStoredFieldsWriter.INDEX_EXTENSION_PREFIX;
```

Comment:
@jpountz I am not sure about this change, and it seems the extension name has been changed. I think there are failures related with this change.

Reply:
Can you try this patch?

```patch
diff --git a/x-pack/plugin/core/src/main/java/org/elasticsearch/snapshots/SourceOnlySnapshot.java b/x-pack/plugin/core/src/main/java/org/elasticsearch/snapshots/SourceOnlySnapshot.java
index 25f8a73..8959a77 100644
--- a/x-pack/plugin/core/src/main/java/org/elasticsearch/snapshots/SourceOnlySnapshot.java
+++ b/x-pack/plugin/core/src/main/java/org/elasticsearch/snapshots/SourceOnlySnapshot.java
@@ -44,15 +44,13 @@ import java.io.ByteArrayOutputStream;
 import java.io.IOException;
 import java.io.PrintStream;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.function.Supplier;
 
-import static org.apache.lucene.codecs.compressing.CompressingStoredFieldsWriter.FIELDS_EXTENSION;
-import static org.apache.lucene.codecs.compressing.CompressingStoredFieldsWriter.INDEX_EXTENSION_PREFIX;
-
 public class SourceOnlySnapshot {
     private final Directory targetDirectory;
     private final Supplier<Query> deleteByQuerySupplier;
@@ -206,14 +204,18 @@ public class SourceOnlySnapshot {
             FieldInfos newFieldInfos = new FieldInfos(fieldInfoCopy.toArray(new FieldInfo[0]));
             codec.fieldInfosFormat().write(trackingDir, newSegmentInfo, segmentSuffix, newFieldInfos, IOContext.DEFAULT);
             newInfo.setFieldInfosFiles(trackingDir.getCreatedFiles());
-            String idxFile = IndexFileNames.segmentFileName(newSegmentInfo.name, segmentSuffix, INDEX_EXTENSION_PREFIX);
-            String dataFile = IndexFileNames.segmentFileName(newSegmentInfo.name, segmentSuffix, FIELDS_EXTENSION);
+            String idxFile = IndexFileNames.segmentFileName(newSegmentInfo.name, segmentSuffix, "fdx");
+            String dataFile = IndexFileNames.segmentFileName(newSegmentInfo.name, segmentSuffix, "fdt");
+            String metaFile = IndexFileNames.segmentFileName(newSegmentInfo.name, segmentSuffix, "fdm"); // new since Lucene 8.5
             Directory sourceDir = newSegmentInfo.dir;
             if (si.getUseCompoundFile()) {
                 sourceDir = codec.compoundFormat().getCompoundReader(sourceDir, si, IOContext.DEFAULT);
             }
             trackingDir.copyFrom(sourceDir, idxFile, idxFile, IOContext.DEFAULT);
             trackingDir.copyFrom(sourceDir, dataFile, dataFile, IOContext.DEFAULT);
+            if (Arrays.asList(sourceDir.listAll()).contains(metaFile)) { // only exists for Lucene 8.5+ indices
+                trackingDir.copyFrom(sourceDir, dataFile, metaFile, IOContext.DEFAULT);
+            }
             if (sourceDir != newSegmentInfo.dir) {
                 sourceDir.close();
             }
```